### DISCUSSION TITLE
How to keep only one dropdown open using useDisclosure in Chakra UI

### DISCUSSION THREAD

**Discussion by PrimozRome** [chakra-ui/chakra-ui#7210] [Q&A]
I am building a main navigation bar dropdown. I am using useDisclosure on each menu item that has a dropdown. I am toggling dropdown on click. I would like that only 1 dropdown is opened at a time. Currently the dropdowns are staying opened. 

[Here is Codesandbox link of my working version.](https://codesandbox.io/p/sandbox/bold-tree-ileh87?file=%2Fcomponents%2FNavItem.js&selection=%5B%7B%22endColumn%22%3A9%2C%22endLineNumber%22%3A32%2C%22startColumn%22%3A9%2C%22startLineNumber%22%3A32%7D%5D)

**Comment by keithburgie** [comment]
The individual `NavItem` components don't have awareness of the others, so you need to manage their state from a shared parent.

I tried to make it work with `useDisclosure()` but I don't think it's the right use case for it. Instead, I've done the following:
1. Create an `activeMenuItem` state in the index file.
2. Create a click handler that compares the `activeMenuItem` with the item that's just been clicked.
-- If it matches, it means that dropdown is already open, so close it. 
-- If it doesn't match, then it's a different menu item, so set that item as active.
3. The `navItem` whose name matches the `activeMenuItem` will be open.

[CodeSanbox fork here](https://codesandbox.io/p/sandbox/chakraui-menu-dropdown-forked-9sm4om?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A24%2C%22endLineNumber%22%3A7%2C%22startColumn%22%3A10%2C%22startLineNumber%22%3A7%7D%5D).
